Braze-Demo-LAB-Progress-Content-Extension/ViewController/NotificationViewController.swift

```diff
@@ -0,0 +1,94 @@
+import UIKit
+import UserNotifications
+import UserNotificationsUI
+
+class NotificationViewController: UIViewController, UNNotificationContentExtension {
+
+  // MARK: - Outlets
+  @IBOutlet private weak var stackView: UIStackView!
+  @IBOutlet private weak var nextSessionLabel: UILabel!
+  @IBOutlet private weak var completeNextSessionByLabel: UILabel!
+  
+  // MARK: - Variables
+  private var sessionData: SessionData! {
+    didSet {
+      rowCount = (sessionData.totalSessionCount + columnCount - 1) / columnCount
+    }
+  }
+  private var rowCount: Int = 0
+  private var columnCount: Int {
+    return traitCollection.verticalSizeClass == .compact ? 6 : 4
+  }
+  
+  func didReceive(_ notification: UNNotification) {
+    let userInfo = notification.request.content.userInfo
+    
+    guard let completedSessionsString = userInfo[PushNotificationKey.completedSessionCount.rawValue] as? String,
+          let totalSessionString = userInfo[PushNotificationKey.totalSessionCount.rawValue] as? String,
+          let completedSessions = Int(completedSessionsString),
+          let totalSessions = Int(totalSessionString)
+    else { fatalError("Key-Value Pairs are incorrect.")}
+    
+    self.sessionData = SessionData(totalSessionCount: totalSessions, completedSessionCount: completedSessions)
+    
+    configureNextSessionLabels(with: userInfo)
+    configureTotalSessionProgress(rows: rowCount, columns: columnCount)
+    
+    saveViewedSessionProgressEvent()
+  }
+}
+
+// MARK: - Private methods
+private extension NotificationViewController {
+  /// Configures the completed session progress layout and views based on the total number of sessions and the number of completed sessions.
+  ///
+  /// The `rowCount` is determined by the total number of sessions the user has and the `columnCount` value. If there are 8 total sessions and the `columnCount` is 6, the layout will be two horizontal `UIStackView`s with 6 sessions each in the vertical `UIStackView`.
+  ///
+  /// If there are not enough sessions to fill the columns in the row, the for loop will create blank views to maintain the UI consistency within the `UIStackView`.
+  func configureTotalSessionProgress(rows: Int, columns: Int) {
+    var currentSessionCount = 1
```

Comment:
It looks like this variable is doing double duty as an index as well as a session `number`. I also don't think the `count` name makes sense here.

Imo it would make sense to cleanly separate the various concepts (you'll also have fewer wonky adjustments for +1 errors). You could have something like:
1. Make everything in the code zero-based - only use one-based values in the UI or some other very limited place (e.g. another field in the Session object for the UI to use)
2. Only call things that refer to sums `counts`
3. Same thing for indices, etc